src/main/java/thaumicenergistics/common/tiles/TileAdvancedInfusionProvider.java

```diff
@@ -0,0 +1,195 @@
+
+/**
+ * Upgrade version of {@link TileInfusionProvider} works like intercepter and old provider can auto order essential it
+ * need
+ * 
+ * @author MCTBL
+ * 
+ */
+public class TileAdvancedInfusionProvider extends TileInfusionProvider implements IAspectSource {
+
+    private static final String NBT_MATRIX_X = "MatrixX", NBT_MATRIX_Y = "MatrixY", NBT_MATRIX_Z = "MatrixZ";
+
+    public TileInfusionMatrix matrix = null;
+
+    private Integer matrixX = null, matrixY = null, matrixZ = null;
+
+    private int tickCounter = 0;
+
+    /**
+     * How much power does this require just to be active?
+     */
+    @Override
+    protected double getIdlePowerusage() {
+        return 50d;
+    }
+
+    @Override
+    protected ItemStack getItemFromTile(final Object obj) {
+        // Return the itemstack the visually represents this tile
+        return ThEApi.instance().blocks().AdvancedInfusionProvider.getStack();
+    }
+
+    @Override
+    public boolean takeFromContainer(final Aspect tag, final int amount) {
+        // Not in advanced mode
+        if (this.matrix == null) {
+            // Can we extract the essentia from the network?
+            if (this.extractEssentiaFromNetwork(tag, amount, true) == amount) {
+                // Show partical FX
+                this.doParticalFX(tag.getColor());
+
+                return true;
+            } else {
+                this.orderSomeEssentia(tag);
+            }
+        }
+        return false;
+    }
+
+    public void searchMatrix() {
+
+        // If allready bind infusion matrix or worldObj is null
+        if (this.matrix != null || this.worldObj == null) {
+            // Do nothing
+            return;
+        }
+
+        int x = this.xCoord;
+        int y = this.yCoord;
+        int z = this.zCoord;
+        for (int dx = -4; dx <= 4; dx++) {
+            for (int dy = -4; dy <= 4; dy++) {
+                for (int dz = -4; dz <= 4; dz++) {
+                    if (dx == 0 && dy == 0 && dz == 0) continue;
+                    if (this.bindMatrix(x + dx, y + dy, z + dz)) {
+                        this.matrixX = x + dx;
+                        this.matrixY = y + dy;
+                        this.matrixZ = z + dz;
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    public boolean bindMatrix(final int x, final int y, final int z) {
+        if (this.worldObj != null && this.worldObj.getTileEntity(x, y, z) instanceof TileInfusionMatrix tim) {
+            this.matrix = tim;
+
+            this.markForUpdate();
+            this.saveChanges();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public boolean bindMatrix() {
+        return this.bindMatrix(this.matrixX, this.matrixY, this.matrixZ);
+    }
+
+    public void unbindMatrix() {
+        this.matrix = null;
+        this.matrixX = null;
+        this.matrixY = null;
+        this.matrixZ = null;
```

Comment:
I missed this previously, but after unbinding matrix all necessary values should be synced and saved to NBT. So `markForUpdate` and `saveChanges` methods should be called here.

Reply:
Not your fault, I add this after your review😂
I will fix this later